src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -241,26 +178,159 @@ std::string postQueryContext
   char portV[16];
   snprintf(portV, sizeof(portV), "%d", port);
 
-  // Fill in the response from the redirection into the response to the originator of this request
-  QueryContextResponse* qcrsP = &parseDataP->qcrs.res;
+  // Fill in the response from the redirection into the response
+  qcrsP->fill(&parseData.qcrs.res);
+  qcrsP->present("", "queryForward, after filling qcrsP");
 
   //
-  // Returning 'redirected to' in StatusCode::details
+  // 'Fixing' StatusCode
   //
-  if ((qcrsP->errorCode.code == SccOk) || (qcrsP->errorCode.details == ""))
+  if (qcrsP->errorCode.code == SccNone)
   {
-    qcrsP->errorCode.details = std::string("Redirected to context provider ") + ip + ":" + portV + prefix;
+    qcrsP->errorCode.fill(SccOk);
   }
-  else if (qcrsP->errorCode.code == SccNone)
+
+  if ((qcrsP->contextElementResponseVector.size() == 1) && (qcrsP->contextElementResponseVector[0]->statusCode.code == SccContextElementNotFound))
   {
-    qcrsP->errorCode.fill(SccOk, std::string("Redirected to context provider ") + ip + ":" + portV + prefix);
+    qcrsP->errorCode.fill(SccContextElementNotFound);
   }
+}
+
+
 
-  if (qcrsP->contextElementResponseVector.size() > 0)
+/* ****************************************************************************
+*
+* postQueryContext -
+*/
+std::string postQueryContext
+(
+  ConnectionInfo*            ciP,
+  int                        components,
+  std::vector<std::string>&  compV,
+  ParseData*                 parseDataP
+)
+{
+  //
+  // Convops calling this routine may need the response in digital
+  // So, the digital response is passed back in parseDataP->qcrs.res
+  //
+  QueryContextResponse*       qcrsP = &parseDataP->qcrs.res;
+  std::string                 answer;
+  int                         noOfForwards = 0;
+  QueryContextRequestVector   requestV;
+  QueryContextResponseVector  responseV;
+
+  qcrsP->errorCode.fill(SccOk);
+  ciP->httpStatusCode = mongoQueryContext(&parseDataP->qcr.res, qcrsP, ciP->tenant, ciP->servicePathV, ciP->uriParam);
+  qcrsP->present("", "postQueryContext");
+
+  for (unsigned int ix = 0 ; ix < qcrsP->contextElementResponseVector.size(); ++ix)
+  {
+    ContextElementResponse* cerP       = qcrsP->contextElementResponseVector[ix]->clone();
+
+    for (unsigned int aIx = 0 ; aIx < cerP->contextElement.contextAttributeVector.size(); ++aIx)
+    {
+      EntityId*            eP       = &cerP->contextElement.entityId;
+      ContextAttribute*    aP       = cerP->contextElement.contextAttributeVector[aIx];
+      QueryContextRequest* requestP = requestV.lookup(aP->providingApplication, eP);   // No need to lookup if providingApplication == "" ...
+      
+      //
+      // An empty providingApplication means the attribute is local
+      // In such a case, the response is already in our hand, we just need to forward it to responseV
```

Comment:
by "forward it to responseV" you mean pushing back into that vector? In that case, "forward" can be a bit confusing, as it is associated to forwarding *in a new REST call* in this context...